src/fu-util.c

```diff
@@ -463,6 +463,179 @@ fu_util_perhaps_show_unreported (FuUtilPrivate *priv, GError **error)
+				"updates.");
+	return str;
+}
+
+static GString *
+fu_util_get_component_agreement (FuUtilPrivate *priv, AsApp *app, GError **error)
+{
+	g_autofree gchar *tmp = NULL;
+#if AS_CHECK_VERSION(0,7,8)
+	AsAgreement *agreement;
+	AsAgreementSection *section;
+
+	/* get the default agreement section */
+	agreement = as_app_get_agreement_default (app);
+	if (agreement == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No agreement found");
+		return NULL;
+	}
+	section = as_agreement_get_section_default (agreement);
+	if (section == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No default section for agreement found");
+		return NULL;
+	}
+	tmp = as_agreement_section_get_description (section, NULL);
+	if (tmp == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No description found in agreement section");
+		return NULL;
+	}
+
+	/* convert to console text */
+	tmp = as_markup_convert_simple (tmp, error);
+#else
+	AsFormat *format;
+	GNode *n;
+	g_autoptr(AsNode) root = NULL;
+	g_autoptr(GFile) file = NULL;
+	g_autoptr(GString) str = NULL;
+
+	/* parse the XML file */
+	format = as_app_get_format_default (app);
+	if (format == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No format for Metainfo file");
+		return NULL;
+	}
+	file = g_file_new_for_path (as_format_get_filename (format));
+	root = as_node_from_file (file, AS_NODE_FROM_XML_FLAG_NONE, NULL, error);
+	if (root == NULL)
+		return NULL;
+
+	/* manually find the first agreement section */
+	n = as_node_find (root, "component/agreement/agreement_section/description");
+	if (n == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No agreement description found");
+		return NULL;
+	}
+	str = as_node_to_xml (n->children, AS_NODE_TO_XML_FLAG_INCLUDE_SIBLINGS);
+	tmp = as_markup_convert_simple (str->str, error);
+#endif
+	return g_string_new (tmp);
+}
+
+static gboolean
+fu_util_modify_remote_warning (FuUtilPrivate *priv, FwupdRemote *remote, GError **error)
+{
+	AsApp *app;
+	g_autofree gchar *component_id = NULL;
+	g_autoptr(AsStore) store = NULL;
+	g_autoptr(GHashTable) os_release = NULL;
+	g_autoptr(GString) desc_plain_str = NULL;
+
+	/* try to find a custom agreement, falling back to a generic warning */
+	store = as_store_new ();
+	as_store_add_filter (store, AS_APP_KIND_SOURCE);
+	if (!as_store_load (store, AS_STORE_LOAD_FLAG_APPDATA, NULL, error))
+		return FALSE;
+	component_id = g_strdup_printf ("org.freedesktop.fwupd.remotes.%s",
+					fwupd_remote_get_id (remote));
+	app = as_store_get_app_by_id (store, component_id);
+	if (app != NULL) {
+		desc_plain_str = fu_util_get_component_agreement (priv, app, error);
+		if (desc_plain_str == NULL)
+			return FALSE;
+	} else {
```

Comment:
This implicitly means that all remotes will show a warning, even if they are intranet remotes.  Is that expected behavior in your mind?  I would think the agreements are important for LVFS but some remotes won't need this fallback model.

Reply:
Good catch, fixed.